### DISCUSSION TITLE
Retrieve custom attributes after creating a user

### DISCUSSION THREAD

**Discussion by mahesh-maney** [Sunbird-Lern/Community#153] [Issues] [answered]
Hi,
We are using  
/api/user/v1/sso/create' and the payload body as 
{
    "request": {
        "firstName": "nodal1 officer1",
        "email": "nodal.officer@yopmail.com",
        "password": "Password@123",
        "lastname": "officer",
        "emailVerified": true,
        "phoneNumber": "9999999999",
        "userName": "nodal.officer",
        "channel": "***hrh***",
        "roles": [
            "CONTENT_CREATOR"
        ]
    }
}'

to create an user. We are able to create an user. However, when we use /api/user/v1/search to fetch user info [ pay load : {
    "request": {
        "filters": {
             "userId": "0ec3136b-4ab7-4ef8-a07d-6ee210a139e6"
        },
        "offset": 0,
        "limit": 10,
        "fields": []
    }
}]
we are not able to fetch the custom attributes (used while creating an user. for Example: phoneNumber, Roles,). I would like to fetch an unmasked email too.

Can you please help me here.

@AmiableAnil @Jayaprakash8887  @BharathwajShankar

**Comment by Jayaprakash8887** [comment]
@mahesh-maney Documents to be indexed to ES has to be updated to show the necessary fields:
 https://github.com/project-sunbird/sunbird-devops/blob/release-5.3.0-lern/ansible/roles/es-mapping/files/mappings/userv3-mapping.json

However, It is not suggested to expose user personal information via search API which could be easily misused. 

Code changes would be required to index non-encrypted email and phone data in  UserServiceImpl.getUserDetailsForES method 

User create flow pertaining to ES: SSOUserCreatorActor.processSSOUser -> BackgroundJobManager.updateUserInfoToEs   -> UserServiceImpl.getUserDetailsForES

**Reply by Jayaprakash8887** [reply, marked as answer]
@mahesh-maney Please mark the thread as answered if there are no further queries on the topic

**Comment by mahesh-maney** [comment]
Thank you @Jayaprakash8887 sir.
